cpp/tensorrt_llm/thop/llama4MinLatency.cpp

```diff
@@ -0,0 +1,219 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-2025 NVIDIA CORPORATION &
+ * AFFILIATES. All rights reserved. SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "tensorrt_llm/common/cudaUtils.h"
+#include "tensorrt_llm/common/dataType.h"
+#include "tensorrt_llm/common/opUtils.h"
+#include "tensorrt_llm/kernels/llama4MinLatencyKernels/llama4Bf16Bf16Gemm.h"
+#include "tensorrt_llm/kernels/llama4MinLatencyKernels/llama4Fp8Bf16Gemm.h"
+#include "tensorrt_llm/kernels/llama4MinLatencyKernels/llama4Fp8Fp8GemmSwiGLU.h"
+#include "tensorrt_llm/kernels/llama4MinLatencyKernels/llama4MinLatencyMoEOp.h"
+#include "tensorrt_llm/runtime/torchUtils.h"
+#include "tensorrt_llm/runtime/utils/mpiUtils.h"
+#include "tensorrt_llm/thop/thUtils.h"
+
+#include <nvml.h>
+#include <torch/extension.h>
+
+#include <cstddef>
+#include <cstdint>
+#include <unordered_set>
+
+namespace torch_ext
+{
+
+torch::Tensor llama4_bf16_bf16_gemm(torch::Tensor inputA, torch::Tensor inputB)
+{
+    CHECK_INPUT(inputA, c10::ScalarType::BFloat16);
+    CHECK_INPUT(inputB, c10::ScalarType::BFloat16);
+
+    TORCH_CHECK(inputA.dim() == 2, "inputA must be 2D.");
+    TORCH_CHECK(inputB.dim() == 2, "inputB must be 2D.");
+    TORCH_CHECK(inputA.size(1) == 5120, "inputA.size(1) must be 5120");
+    TORCH_CHECK(inputB.size(0) == 128, "inputB.size(0) must be 128");
+    TORCH_CHECK(inputB.size(1) == 5120, "inputB.size(1) must be 5120");
```

Comment:
```suggestion
    TORCH_CHECK(inputA.size()[1] == 5120, "inputA.size(1) must be 5120");
    TORCH_CHECK(inputB.size()[0] == 128, "inputB.size(0) must be 128");
    TORCH_CHECK(inputB.size()[1] == 5120, "inputB.size(1) must be 5120");
```
Tensor.size(i) is less efficient because it goes through the pytorch dispatcher. Please update the rest of the file.